R/Backend-class.R

```diff
@@ -249,3 +249,23 @@ setGeneric(
         standardGeneric("backendWriteSpectra"),
     valueClass="Backend"
 )
+
+#' Subset the `Backend` by spectra (`i`) and/or file (`j`). Note that the
+#' default implementation ignores argument `i`.
+#'
+#' @param x `Backend`
+#'
+#' @param i `integer`; ignored.
+#'
+#' @param j `integer` defining the files to which `x` should be subsetted.
+#'
+#' @return A `Backend` class.
+#'
+#' @author Johannes Rainer
+#'
+#' @rdname hidden_aliases
+setMethod("[", "Backend", function(x, i, j, ..., drop = TRUE) {
```

Comment:
See discussion in #417. Maybe we should not ignore `i` (global spectra index (file crossing)) but `j` and provide an additional method `filterFile`.

Reply:
Please note: I just moved this myself to `filterFile` because I wanted it for the `BackendMemory` class.

Reply:
I would not use `filterFile,Backend` and `[,Backend`. See https://github.com/lgatto/MSnbase/issues/417#issuecomment-460157900